### DISCUSSION TITLE
PingVin LXC update failed

### DISCUSSION THREAD

**Discussion by Narkowii** [community-scripts/ProxmoxVE#50] [Q&A] [answered]
hello
i ve got an failed update with the script

`✓ Stopped Pingvin Share
 -remote: Enumerating objects: 56, done.
remote: Counting objects: 100% (56/56), done.
remote: Compressing objects: 100% (47/47), done.
remote: Total 56 (delta 28), reused 20 (delta 9), pack-reused 0 (from 0)
Unpacking objects: 100% (56/56), 52.50 KiB | 1.94 MiB/s, done.
From https://github.com/stonith404/pingvin-share
 + 6bc6271...0ca0191 i18n_crowdin -> origin/i18n_crowdin  (forced update)
   ab4f19e..77eef18  main         -> origin/main
 \
[ERROR] in line 68: exit code 0: while executing command npm run build &> /dev/null`

**Comment by remz1337** [comment]
NPM can fail if it's missing resources. Try to increase RAM/CPU. Otherwise, maybe there has been a breaking change in a recent PingVin release. Would need to investigate

**Comment by Narkowii** [comment]
i have increase RAM and CPU....same error

**Comment by b3nw** [comment]
also run into trouble on update

```
✓ Stopped Pingvin Share
 /remote: Enumerating objects: 1945, done.
remote: Counting objects: 100% (1060/1060), done.
remote: Compressing objects: 100% (487/487), done.
remote: Total 1945 (delta 754), reused 762 (delta 538), pack-reused 885 (from 1)
Receiving objects: 100% (1945/1945), 3.03 MiB | 17.07 MiB/s, done.
Resolving deltas: 100% (1238/1238), completed with 23 local objects.
 -From https://github.com/stonith404/pingvin-share
 * [new branch]      gh-pages                  -> origin/gh-pages
 + 986c1c4...b470561 i18n_crowdin              -> origin/i18n_crowdin  (forced update)
 * [new branch]      log-invalid-oidc-response -> origin/log-invalid-oidc-response
   414bcec..51478b6  main                      -> origin/main
 * [new tag]         v0.28.0                   -> v0.28.0
 * [new tag]         v0.29.0                   -> v0.29.0
 * [new tag]         v1.0.0                    -> v1.0.0
 * [new tag]         v1.0.1                    -> v1.0.1
 * [new tag]         v1.0.2                    -> v1.0.2
 * [new tag]         v1.0.3                    -> v1.0.3
 * [new tag]         v1.0.4                    -> v1.0.4
 * [new tag]         v1.1.0                    -> v1.1.0
 * [new tag]         v1.1.1                    -> v1.1.1
 * [new tag]         v1.1.2                    -> v1.1.2
 * [new tag]         v1.1.3                    -> v1.1.3
 * [new tag]         v1.2.0                    -> v1.2.0
 * [new tag]         v1.2.1                    -> v1.2.1
 * [new tag]         v1.2.2                    -> v1.2.2
 * [new tag]         v1.2.3                    -> v1.2.3
 * [new tag]         v1.2.4                    -> v1.2.4
 * [new tag]         v1.3.0                    -> v1.3.0
 |
[ERROR] in line 65: exit code 0: while executing command git checkout $(git describe --tags `git rev-list --tags --max-count=1`) &> /dev/null
```

**Comment by Narkowii** [comment, marked as answer]
managed to do the update like this

cd /opt/pingvin-share

pm2 stop pingvin-share-backend pingvin-share-frontend

cd backend
npm install
npm run build
pm2 start --name="pingvin-share-backend" npm -- run prod

cd ../frontend
npm install
npm run build
API_URL=http://192.168.x.x:3000 # Set the URL of the backend
pm2 start npm --name "pingvin-share-frontend" -- run start
